navapbc/template-application-rails
# Docker Console and Debugger

## Ticket

- Resolves #14 

## Changes

- Updated web_console config to allow most local IPs to access the console for local development with docker
- Updated docs to give more specific instructions on how to access the debugger

## Questions for reviewers

1. In the technical-foundation doc, it says to run the app with `make rails-console` though I think elsewhere it says to use `make init-native/container` and `make start-native/container` for local native and local docker development. Should we change it here to be consistent?
2. We're pulling the ruby:3.3.1-slim image from dockerhub which comes with the debug gem version 1.19.1. However, in our local gemfile.lock we have debug version 1.19.2 installed. Using debug in docker breaks when there is a version mismatch between the server and the client. This will be fixed whenever dockerhub updates their ruby:3.3.1-slim image, but I was only able to get it running locally by locking my debug version in the gemfile to 1.9.1. Is there a best practice to address this?
3. When setting the IP addresses for permissions, there are some programmatic ways of getting locaL IP addresses, but it seems like those require additional libraries (socket, ipaddr). Is that preferred to the current method or should we go even further by allowing `0.0.0.0`?

## Testing

- Add a console statement to a view in the application and confirm that the console shows up in your browser when hitting that page. Test in both native and container modes.
- Add a debugger statement to a controller and follow the instructions in the technical-foundation.md doc to attach the debugger. Confirm in both native and container modes.
